### PR TITLE
Clarify the documentation of the GridIn class.

### PR DESCRIPTION
**Pull request by bangerth**
This is in response to a question on the mailing list today.

### CODE REVIEW

File: include/deal.II/grid/grid_in.h [hunk 1]
```diff
@@ -65,25 +68,19 @@ template <int dim> struct CellData;
...
- * numbering in the input file, which need not be consecutively, is lost upon
- * transfer to the triangulation object, since this one needs consecutively
- * numbered elements.
- *
- * Material indicators are accepted to denote the material ID of cells and to
- * denote boundary part indication for lines in 2D. Read the according
- * sections in the documentation of the Triangulation class for further
- * details.
+ * To read grid data, the triangulation to be filled has to be empty.
+ * Upon calling the functions of this class, the input file may only lines in
```
Comment (tjhei) [inline include/deal.II/grid/grid_in.h]: missing "contain" or "consist of"?

File: include/deal.II/grid/grid_in.h [hunk 2]
```diff
@@ -65,25 +68,19 @@ template <int dim> struct CellData;
...
+ * one dimension; lines and quads in two dimensions; and lines, quads, and
+ * hexes in three dimensions. All other cell types (e.g. triangles
+ * in two dimensions, triangles or tetrahedra in 3d) are rejected. The result
```
Comment (tjhei) [inline include/deal.II/grid/grid_in.h]: can you discuss the codim=1 and 2 case here too?

File: include/deal.II/grid/grid_in.h [hunk 3]
```diff
@@ -65,25 +68,19 @@ template <int dim> struct CellData;
...
+ * will be a triangulation that consists of the cells described in the input
+ * file, and to the degree possible with material indicators and boundary
+ * indicators correctly set as described in the input file. On the other
+ * hand, you can not expect vertex and cell numbers in the triangulation
```
Comment (tjhei) [inline include/deal.II/grid/grid_in.h]: "on the other hand" of what?

**Comment by bangerth** [comment]
All good points. Updated.

**Pull request merged**